cluster/kubernetes/sync.go

```diff
@@ -72,22 +363,45 @@ func (c *Kubectl) connectArgs() []string {
 	return args
 }
 
+// getDefaultNamespace returns the fallback namespace used by the
+// applied when a namespaced resource doesn't have one specified. This
+// is used when syncing to anticipate the identity of a resource in
+// the cluster given the manifest from a file (which may be missing
+// the namespace).
+func (k *Kubectl) getDefaultNamespace() string {
```

Comment:
I think we should test that this function works with the output of the `kubectl` binary we ship with `fluxd` (in case it changes).

Maybe there are integration to test this already, but, from https://github.com/weaveworks/flux/pull/1101 it seems that's not the case.

Reply:
Yeah -- this deserves a regression test at least.

Reply:
Happy to create a test for this after merging #1765 and rebasing on top of it.